Add .discard() method. (#4636)
Codecademy/docs: content/python/concepts/sets/terms/discard/discard.md

---
Title: '.discard()'
Description: 'Removes a passing element from the set if it's present.'
Subjects: 
  - 'Computer Science'
  - 'Data Science'
Tags: 
  - 'Collections'
  - 'Functions'
  - 'Methods'
  - 'Sets'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The `.discard()` method removes the specified item from the set of objects. 

## Syntax

The syntax is the following:

```
set_name.discard(value)
```

> - This method takes exactly one argument;

Reviewer: This line can be removed, instead of this line describe the parameters in the following format:
- `set`: Parameter description
- `value`: Parameter description

> - It differs from the `.remove()` method because it **will not raise a KeyError** if the specified item doesn't exist in the set.
> 

Reviewer: ```suggestion
```

## Example

The example below shows possible usage of the method:

```py

> coffee_set = {'espresso', 'flat_white', 'cappuccino', 'filter'}
> print(coffee_set)
> 
> coffee_set.discard('espresso')
> print(coffee_set)
> 
> coffee_set.discard('aeropress')
> print(coffee_set)

Reviewer: ```suggestion
coffee_set = {'espresso', 'flat_white', 'cappuccino', 'filter'}
print(coffee_set)

# removing 'espresso' from the set 
coffee_set.discard('espresso') 
print(coffee_set)

# removing 'aeropress' from the set 
coffee_set.discard('aeropress') 
print(coffee_set)
```

```

> The output would be the following:

Reviewer: ```suggestion
The code above produces  the following output:
```

> ```py

Reviewer: ```suggestion
```shell
```

{'espresso', 'flat_white', 'cappuccino', 'filter'}
{'flat_white', 'cappuccino', 'filter'}
{'flat_white', 'cappuccino', 'filter'}
```

## Codebyte Example

> In the example, we see that our code continues to execute though there are already no coffee beans from Vietnam in our set.

Reviewer: ```suggestion
In the example, our code continues to execute without errors even though we attempt to remove elements ('Vietnam' and 'Brazil') from the set coffee_beans that are not present in the set.
```

```codebyte/python
coffee_beans = {'Brazil', 'Ethiopia', 'Kenya', 'Columbia'}
print(coffee_beans)

coffee_beans.discard('Vietnam')
coffee_beans.discard('Brazil')
print(coffee_beans)

> ```

Reviewer: ```suggestion
```

> **Note:** The key difference between `.discard()` and `.remove()` in Python's set is that `.discard()` doesn't raise an error if the element is not present, while `.remove()` raises a KeyError in such a case.
```
